src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrResourceProvider.java

```diff
@@ -478,6 +479,44 @@ public Resource create(final @NotNull ResolveContext<JcrProviderState> ctx, fina
         }
     }
 
+    @Override
+    public boolean orderBefore(@NotNull ResolveContext<JcrProviderState> ctx, @NotNull Resource parent, @NotNull String name,
+            @Nullable String followingSiblingName) throws PersistenceException {
+        Node node = parent.adaptTo(Node.class);
+        try {
+            if (node == null) {
+                throw new PersistenceException("The resource " + parent.getPath() + " cannot be adapted to Node. It is probably not provided by the JcrResourceProvider");
+            }
+            // check if reordering necessary
+            NodeIterator nodeIterator = node.getNodes();
+            long existingNodePosition = -1;
+            long index = 0;
+            while (nodeIterator.hasNext()) {
+                Node childNode = nodeIterator.nextNode();
+                if (childNode.getName().equals(name)) {
+                    existingNodePosition = index;
+                }
+                if (existingNodePosition >= 0) {
```

Comment:
Does this loop assume that the node named `name` comes first in the children list? I am having a bit of a hard time following it.

Reply:
No, not necessarily. I added some tests. Hopefully they proof that `true` is only returned in case the order is actually changed. Let me know if I missed a case.